keystorage/yubihsm2/src/main/java/tech/pegasys/signers/yubihsm2/YubiHsm2.java

```diff
@@ -0,0 +1,120 @@
+ * uses yubihsm-shell, which is a thin wrapper over libyubihsm.
+ */
+public class YubiHsm2 {
+  private final List<String> yubiHsmShellPathArgs;
+  private final String connectorUrl;
+  private final short authKey;
+  private final String password;
+  private final Optional<String> caCert;
+  private final Optional<String> proxy;
+  private final Optional<Map<String, String>> additionalEnvVars;
+
+  /**
+   * Parameters that will be passed to yubihsm-shell process to fetch opaque data
+   *
+   * @param yubiHsmShellPathArgs Path to yubihsm-shell binary and any additional arguments to pass.
+   * @param additionalEnvVars Optional environment variables map that can be passed to yubihsm-shell
+   * @param connectorUrl YubiHSM connector URL. http://127.0.01:12345 or yhusb://serial=123456
+   * @param authKeyObjId The auth key object id
+   * @param password Password
+   * @param caCert Optional CA Cert if connector is running in https mode
+   * @param proxy Optional proxy url if a proxy should be used to access connector url
+   */
+  public YubiHsm2(
+      final List<String> yubiHsmShellPathArgs,
+      final Optional<Map<String, String>> additionalEnvVars,
+      final String connectorUrl,
+      final short authKeyObjId,
+      final String password,
+      Optional<String> caCert,
+      Optional<String> proxy) {
+    this.yubiHsmShellPathArgs = yubiHsmShellPathArgs;
+    this.additionalEnvVars = additionalEnvVars;
+    this.connectorUrl = connectorUrl;
+    this.authKey = authKeyObjId;
+    this.password = password;
+    this.caCert = caCert;
+    this.proxy = proxy;
+  }
+
+  /**
+   * Fetch opaque data from yubi hsm 2
+   *
+   * @param opaqueObjId The object id of opque
+   * @return The data stored against opaque object id (in hex format)
+   * @throws YubiHsmException In case of errors in invoking process or if command returns error.
+   */
+  public String fetchKey(final short opaqueObjId) throws YubiHsmException {
+    try {
+      return executeProcess(
+          getOpaqueDataArgs(opaqueObjId),
+          additionalEnvVars.orElse(Collections.emptyMap()),
+          password);
+    } catch (final IOException | TimeoutException | InterruptedException e) {
+      throw new YubiHsmException("Error in invoking yubihsm-shell process: " + e.getMessage());
+    }
+  }
+
+  static String executeProcess(
+      final List<String> args, final Map<String, String> additionalEnvVars, String password)
+      throws IOException, TimeoutException, InterruptedException {
+    final ProcessResult processResult =
+        new ProcessExecutor()
+            .environment(additionalEnvVars)
+            .command(args)
+            .readOutput(true)
+            .redirectInput(IOUtils.toInputStream(password + "\n", StandardCharsets.UTF_8.name()))
+            .destroyOnExit()
+            .execute();
+
+    final List<String> linesAsUTF8 = processResult.getOutput().getLinesAsUTF8();
```

Comment:
linesAsUtf8? or what about utf8Lines? (just not capitalised UTF8)

Reply:
auto-gen by IntelliJ. Let me fix it.

Reply:
fixed as outputLines.